soccer_vision_3d_msgs/scripts/visualizer.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+
+import rclpy
+import time
+from rclpy.node import Node
+from rclpy.duration import Duration
+
+from geometry_msgs.msg import Vector3
+from std_msgs.msg import ColorRGBA
+from visualization_msgs.msg import Marker
+from soccer_vision_3d_msgs.msg import BallArray, FieldBoundary, GoalpostArray, RobotArray
+from soccer_vision_attribute_msgs.msg import Confidence, Robot as RobotAttribute
+
+
+class SoccerVision3DMsgs2Rviz(Node):
+    """This node provides RViz markers coresponding to the recognized objects."""
+
+    def __init__(self):
+        super().__init__('SVM2DRviz')
+
+        # Create publishers
+        self.marker_publisher = self.create_publisher(Marker, "soccer_vision_3d_msgs", 1)
+
+        # Settings
+        self.ball_diameter = 0.13
+        self.ball_lifetime = int(5e9)
+        self.goal_lifetime = int(5e9)
+        self.obstacle_lifetime = int(5e9)
+        self.robot_lifetime = int(5e9)
+
+        # Ball
+        self.marker_ball = Marker()
+        self.marker_ball.ns = "rel_ball"
+        self.marker_ball.id = 0
+        self.marker_ball.type = Marker.SPHERE
+        self.marker_ball.action = Marker.MODIFY
+        self.marker_ball.scale = Vector3(** dict(zip("xyz", [self.ball_diameter] * 3)))
+        self.marker_ball.color = ColorRGBA(r=1.0, a=1.0)
+        self.marker_ball.lifetime = Duration(nanoseconds=self.ball_lifetime).to_msg()
+
+        # Goalpost
+        self.marker_goalpost = Marker()
+        self.marker_goalpost.ns = "rel_goal"
+        self.marker_goalpost.id = 0
+        self.marker_goalpost.type = Marker.CYLINDER
+        self.marker_goalpost.action = Marker.MODIFY
+        self.marker_goalpost.scale = Vector3()
```

Comment:
Scale gets set later. This line should be unnecessary.

Reply:
Right